taichi/transforms/type_check.cpp

```diff
@@ -335,6 +335,19 @@ class TypeCheck : public IRVisitor {
     stmt->ret_type = VectorType(1, arg_type);
   }
 
+  void visit(KernelReturnStmt *stmt) {
+    Kernel *current_kernel = kernel;
+    if (current_kernel == nullptr) {
+      current_kernel = &get_current_program().get_current_kernel();
+    }
+    auto &rets = current_kernel->rets;
+    TI_ASSERT(rets.size() >= 1);
+    auto ret = rets[0]; // TODO: stmt->ret_id?
```

Comment:
```suggestion
    // TODO: Support cases when stmt->ret_id other than 0
    TI_ASSERT(stmt->ret_id == 0);
    auto ret = rets[0]; 
```

Reply:
It's better to let unsupported cases fail loudly.

Reply:
Sorry but we don't have `ret_id` yet. Will add in another PR.